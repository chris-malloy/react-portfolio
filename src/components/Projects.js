import React from 'react';
import { projectArray } from '../constants';
import Project from './Project';
import { Row } from 'react-materialize';

const Projects = () => {
  const projectList = projectArray.map((p, i) => {
    return <Project key={i} title={p.title} href={p.href} href2={p.href2} link={p.link} link2={p.link2} description={p.description} image={p.image} />
  });
  return (
      <Row>
        {projectList}
      </Row>
    )
  };
  
export default Projects;