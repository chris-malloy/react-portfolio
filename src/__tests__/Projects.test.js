import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Projects from '../components/Projects';

configure({ adapter: new Adapter() });

describe('Projects', () => {
  const projects = shallow(<Projects />);

  it('renders correctly', () => {
    expect(projects).toMatchSnapshot();
  }); 
});