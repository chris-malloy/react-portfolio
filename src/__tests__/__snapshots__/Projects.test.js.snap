// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Projects />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Project
          description="A fun web app that gives recommendations based on current weather."
          href="https://github.com/chris-malloy/weatherU"
          href2="https://weatheru.chrismalloy.net/"
          image="images/weatherU-ss.png"
          link="Github"
          link2="Live Site"
          title="weatherU"
/>,
        <Project
          description="A React Native app that uses Google Maps API to track your running routes."
          href="https://github.com/caseyezimmerman/TrackR"
          href2={undefined}
          image="images/trackr-ss.png"
          link="Github"
          link2={undefined}
          title="Trackr"
/>,
        <Project
          description="An e-commerce site built with React/Redux and Node.js"
          href="https://github.com/chris-malloy/ecommerce"
          href2="https://bittymotors.chrismalloy.net/"
          image="images/bitty-motors-ss.png"
          link="Github"
          link2="Live Site"
          title="Bitty Motors"
/>,
        <Project
          description="This web app utilizes Google Maps API to generate running paths based on their current location and desired distance."
          href="https://github.com/caseyezimmerman/cityRunner"
          href2={undefined}
          image="images/city-runner-ss.png"
          link="Github"
          link2={undefined}
          title="City Runner"
/>,
        <Project
          description="A twist on an old card game. Built with Bootstrap and jQuery."
          href="https://github.com/chris-malloy/tronjack"
          href2={undefined}
          image="images/tronjack-ss.png"
          link="Github"
          link2={undefined}
          title="Tronjack"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "description": "A fun web app that gives recommendations based on current weather.",
          "href": "https://github.com/chris-malloy/weatherU",
          "href2": "https://weatheru.chrismalloy.net/",
          "image": "images/weatherU-ss.png",
          "link": "Github",
          "link2": "Live Site",
          "title": "weatherU",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "description": "A React Native app that uses Google Maps API to track your running routes.",
          "href": "https://github.com/caseyezimmerman/TrackR",
          "href2": undefined,
          "image": "images/trackr-ss.png",
          "link": "Github",
          "link2": undefined,
          "title": "Trackr",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "description": "An e-commerce site built with React/Redux and Node.js",
          "href": "https://github.com/chris-malloy/ecommerce",
          "href2": "https://bittymotors.chrismalloy.net/",
          "image": "images/bitty-motors-ss.png",
          "link": "Github",
          "link2": "Live Site",
          "title": "Bitty Motors",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "description": "This web app utilizes Google Maps API to generate running paths based on their current location and desired distance.",
          "href": "https://github.com/caseyezimmerman/cityRunner",
          "href2": undefined,
          "image": "images/city-runner-ss.png",
          "link": "Github",
          "link2": undefined,
          "title": "City Runner",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "description": "A twist on an old card game. Built with Bootstrap and jQuery.",
          "href": "https://github.com/chris-malloy/tronjack",
          "href2": undefined,
          "image": "images/tronjack-ss.png",
          "link": "Github",
          "link2": undefined,
          "title": "Tronjack",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Project
            description="A fun web app that gives recommendations based on current weather."
            href="https://github.com/chris-malloy/weatherU"
            href2="https://weatheru.chrismalloy.net/"
            image="images/weatherU-ss.png"
            link="Github"
            link2="Live Site"
            title="weatherU"
/>,
          <Project
            description="A React Native app that uses Google Maps API to track your running routes."
            href="https://github.com/caseyezimmerman/TrackR"
            href2={undefined}
            image="images/trackr-ss.png"
            link="Github"
            link2={undefined}
            title="Trackr"
/>,
          <Project
            description="An e-commerce site built with React/Redux and Node.js"
            href="https://github.com/chris-malloy/ecommerce"
            href2="https://bittymotors.chrismalloy.net/"
            image="images/bitty-motors-ss.png"
            link="Github"
            link2="Live Site"
            title="Bitty Motors"
/>,
          <Project
            description="This web app utilizes Google Maps API to generate running paths based on their current location and desired distance."
            href="https://github.com/caseyezimmerman/cityRunner"
            href2={undefined}
            image="images/city-runner-ss.png"
            link="Github"
            link2={undefined}
            title="City Runner"
/>,
          <Project
            description="A twist on an old card game. Built with Bootstrap and jQuery."
            href="https://github.com/chris-malloy/tronjack"
            href2={undefined}
            image="images/tronjack-ss.png"
            link="Github"
            link2={undefined}
            title="Tronjack"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "description": "A fun web app that gives recommendations based on current weather.",
            "href": "https://github.com/chris-malloy/weatherU",
            "href2": "https://weatheru.chrismalloy.net/",
            "image": "images/weatherU-ss.png",
            "link": "Github",
            "link2": "Live Site",
            "title": "weatherU",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "description": "A React Native app that uses Google Maps API to track your running routes.",
            "href": "https://github.com/caseyezimmerman/TrackR",
            "href2": undefined,
            "image": "images/trackr-ss.png",
            "link": "Github",
            "link2": undefined,
            "title": "Trackr",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "description": "An e-commerce site built with React/Redux and Node.js",
            "href": "https://github.com/chris-malloy/ecommerce",
            "href2": "https://bittymotors.chrismalloy.net/",
            "image": "images/bitty-motors-ss.png",
            "link": "Github",
            "link2": "Live Site",
            "title": "Bitty Motors",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "description": "This web app utilizes Google Maps API to generate running paths based on their current location and desired distance.",
            "href": "https://github.com/caseyezimmerman/cityRunner",
            "href2": undefined,
            "image": "images/city-runner-ss.png",
            "link": "Github",
            "link2": undefined,
            "title": "City Runner",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "description": "A twist on an old card game. Built with Bootstrap and jQuery.",
            "href": "https://github.com/chris-malloy/tronjack",
            "href2": undefined,
            "image": "images/tronjack-ss.png",
            "link": "Github",
            "link2": undefined,
            "title": "Tronjack",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
